pyenttec/__init__.py

```diff
@@ -183,36 +189,36 @@ def _update_params(self):
                         (univ_size + 1) & 0xFF,
                         ( (univ_size + 1) >> 8) & 0xFF,
                         0]
-        self._packet_start = ''.join(chr(v) for v in packet_start)
+        self._packet_start = array('B', packet_start).tostring()
 
         self._write_settings()
 
     def _write_settings(self):
         """Write the current settings to the port."""
-        self.com.write(self._port_params.to_packet())
+        self.com.write(self._port_params.to_packet() + _PACKET_END)
```

Comment:
Good catch on the missing _PACKET_END; I think it belongs in the `to_packet` method itself, though.

Reply:
Oops! In-fact I don't think it is so great. `to_packet` already has `_END_VAL` added onto it's array. Perhaps I will change it to use `_PACKET_END` in that method to be consistent.